From e2a5e820424fce5165cb7cba2916e3edbc035504 Mon Sep 17 00:00:00 2001
From: Devaraj K <devaraj@apache.org>
Date: Thu, 9 Jul 2015 15:00:26 +0530
Subject: [PATCH 1419/2748] MAPREDUCE-6426.
 TestShuffleHandler#testGetMapOutputInfo is
 failing. Contributed by zhihai xu.

(cherry picked from commit fffb15bb431fcdd7ca4ce0e249f9d45f4968497b)
(cherry picked from commit 3074d7082e779c1f32c60401887186776a61bdaf)

Change-Id: I21f19cb0820dedd4f70cd8db0733ab404a613623
---
 .../apache/hadoop/mapred/TestShuffleHandler.java   |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-shuffle/src/test/java/org/apache/hadoop/mapred/TestShuffleHandler.java b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-shuffle/src/test/java/org/apache/hadoop/mapred/TestShuffleHandler.java
index 859d276..ff8920f 100644
--- a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-shuffle/src/test/java/org/apache/hadoop/mapred/TestShuffleHandler.java
+++ b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-shuffle/src/test/java/org/apache/hadoop/mapred/TestShuffleHandler.java
@@ -945,6 +945,9 @@ public void testGetMapOutputInfo() throws Exception {
     Configuration conf = new Configuration();
     conf.setInt(ShuffleHandler.SHUFFLE_PORT_CONFIG_KEY, 0);
     conf.setInt(ShuffleHandler.MAX_SHUFFLE_CONNECTIONS, 3);
+    conf.set(CommonConfigurationKeysPublic.HADOOP_SECURITY_AUTHENTICATION,
+        "simple");
+    UserGroupInformation.setConfiguration(conf);
     File absLogDir = new File("target", TestShuffleHandler.class.
         getSimpleName() + "LocDir").getAbsoluteFile();
     conf.set(YarnConfiguration.NM_LOCAL_DIRS, absLogDir.getAbsolutePath());
@@ -1032,7 +1035,8 @@ protected ChannelFuture sendMapOutput(ChannelHandlerContext ctx,
       } catch (EOFException e) {
         // ignore
       }
-      Assert.assertEquals(failures.size(), 0);
+      Assert.assertEquals("sendError called due to shuffle error",
+          0, failures.size());
     } finally {
       shuffleHandler.stop();
       FileUtil.fullyDelete(absLogDir);
-- 
1.7.9.5

